automata/fa/nfa.py

```diff
@@ -81,6 +81,13 @@ def __or__(self, other):
         else:
             raise NotImplementedError
 
+    def __and__(self, other):
+        """Return the union of this NFA and another NFA."""
+        if isinstance(other, NFA):
+            return self.intersection(other)
+        else:
+            raise NotImplementedError
```

Comment:
Maybe change to `return NotImplemented`? Need to update all DFA operators then as well

Reply:
Yeah, I actually thought returning that would be better, but didn't want to update tests before discussion here. @caleb531 what do you think?

Reply:
@eliotwrobson Yes, because `__and__` is a binary magic method, `return NotImplemented` should be used instead. This is the rule of thumb that I follow:

## NotImplemented

> should be returned by the binary special methods (e.g. __eq__(), __lt__(), __add__(), __rsub__(), etc.) to indicate that the operation is not implemented with respect to the other type

## NotImplementedError

> [...] In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method, or while the class is being developed to indicate that the real implementation still needs to be added.

Source: https://stackoverflow.com/a/44575926

Reply:
@eliotwrobson I can take care of the DFA class right now. 🙂

Reply:
@caleb531 Awesome! I think this PR should be good to go now.